src/appleseed/foundation/image/oiioimagefilewriter.cpp

```diff
@@ -0,0 +1,515 @@
+
+    // Format of the pixel data.
+    set_image_output_format(props.m_pixel_format);
+}
+
+void OIIOImageFileWriter::set_exr_image_attributes(const ImageAttributes& image_attributes)
+{
+    OIIO::ImageSpec spec = m_images->m_spec.back();
+
+    if (image_attributes.exist("dwa_compression_lvl"))
+        spec.attribute("openexr:dwaCompressionLevel", image_attributes.get<float>("dwa_compression_lvl"));
+
+    if (image_attributes.exist("white_xy_chromaticity") &&
+        image_attributes.exist("red_xy_chromaticity") &&
+        image_attributes.exist("green_xy_chromaticity") &&
+        image_attributes.exist("blue_xy_chromaticity"))
+    {
+        float chromaticities[8];
+        memset(chromaticities, 0, 8 * sizeof(float));
+
+        const Vector2f red = image_attributes.get<Vector2f>("red_xy_chromaticity");
+        const Vector2f green = image_attributes.get<Vector2f>("green_xy_chromaticity");
+        const Vector2f blue = image_attributes.get<Vector2f>("blue_xy_chromaticity");
+        const Vector2f white = image_attributes.get<Vector2f>("white_xy_chromaticity");
+
+        chromaticities[0] = red[0]; chromaticities[1] = red[1];
+        chromaticities[2] = green[0]; chromaticities[3] = green[1];
+        chromaticities[4] = blue[0]; chromaticities[5] = blue[1];
+        chromaticities[6] = white[0]; chromaticities[7] = white[1];
+
+        OIIO::TypeDesc type;
+        type.basetype = OIIO::TypeDesc::BASETYPE::FLOAT;
+        type.aggregate = OIIO::TypeDesc::AGGREGATE::SCALAR;
+        type.arraylen = 8;
+
+        spec.attribute("chromaticities", type, chromaticities);
+    }
+}
+
+void OIIOImageFileWriter::set_generic_image_attributes(const ImageAttributes& image_attributes)
+{
+    OIIO::ImageSpec spec = m_images->m_spec.back();
+
+    for (const_each<ImageAttributes> i = image_attributes; i; ++i)
+    {
+        // Fetch the name and the value of the attribute.
+        const std::string attr_name = i->key();
+        const std::string attr_value = i->value<std::string>();
+
+        if (attr_name == "author")
+            spec.attribute("Artist", attr_value.c_str());
+
+        else if (attr_name == "copyright")
+            spec.attribute("Copyright", attr_value.c_str());
+
+        else if (attr_name == "title")
+            spec.attribute("DocumentName", attr_value.c_str());
+
+        else if (attr_name == "description")
+            spec.attribute("ImageDescription", attr_value.c_str());
+
+        else if (attr_name == "date")
+            spec.attribute("DateTime", attr_value.c_str());
+
+        else if (attr_name == "software")
+            spec.attribute("Software", attr_value.c_str());
+
+        else if (attr_name == "computer")
+            spec.attribute("HostComputer", attr_value.c_str());
+
+        else if (attr_name == "image_name")
+            spec.attribute("oiio:subimagename", attr_value.c_str());
+
+        else if (attr_name == "color_space")
+        {
+            if (attr_value == "linear")
```

Comment:
Why is there a special case for `linear`?

Reply:
It's the historical value for "linear" of Appleseed.
But OIIO want the first letter as uppercase.

Should I change `linear` by `Linear` in Appleseed ?